src/flush_tracker.rs

```diff
@@ -0,0 +1,316 @@
+    pub fn clear_queues(&self) {
+        self.tasks.clear();
+    }
+
+    /// Gets the names of partitions that have queued tasks.
+    #[inline(always)]
+    pub fn get_partitions_with_tasks(&self) -> Vec<PartitionKey> {
+        self.tasks.get_partitions_with_tasks()
+    }
+
+    /// Returns the amount of queues.
+    #[inline(always)]
+    pub fn queue_count(&self) -> usize {
+        self.tasks.queue_count()
+    }
+
+    /// Returns the amount of bytes queued.
+    #[inline(always)]
+    pub fn queued_size(&self) -> u64 {
+        self.tasks.queued_size()
+    }
+
+    // NOTE: is actually used in tests
+    #[allow(dead_code)]
+    /// Returns the amount of tasks that are queued to be flushed.
+    #[inline(always)]
+    pub fn task_count(&self) -> usize {
+        self.tasks.task_count()
+    }
+
+    // NOTE: is actually used in tests
+    #[allow(dead_code)]
+    #[must_use]
+    #[inline(always)]
+    pub fn is_task_queue_empty(&self) -> bool {
+        self.tasks.is_empty()
+    }
+
+    #[inline(always)]
+    pub fn remove_partition(&self, name: &str) {
+        self.tasks.remove_partition(name);
+    }
+
+    #[inline(always)]
+    pub fn enqueue_task(&self, task: Task) {
+        self.tasks.enqueue(task);
+    }
+
+    /// Returns a list of tasks per partition.
+    #[inline(always)]
+    pub fn collect_tasks(&self, limit: usize) -> Vec<Vec<Arc<Task>>> {
+        self.tasks.collect_tasks(limit)
+    }
+
+    #[inline(always)]
+    pub fn dequeue_tasks(&self, partition_name: &PartitionKey, cnt: usize) {
+        self.tasks.dequeue(partition_name, cnt);
+    }
+}
+
+/// Journal Item Queue
+impl FlushTracker {
+    #[inline(always)]
+    pub fn clear_items(&self) {
+        self.items.clear();
+    }
+
+    /// Returns the amount of journals
+    #[inline(always)]
+    pub fn journal_count(&self) -> usize {
+        self.items.journal_count()
+    }
+
+    // NOTE: is actually used in tests
+    #[allow(dead_code)]
+    /// Returns the amount of sealed journals
+    #[inline(always)]
+    pub fn sealed_journal_count(&self) -> usize {
+        self.items.sealed_journal_count()
+    }
+
+    /// Returns the amount of bytes used on disk by journals
+    #[inline(always)]
+    pub fn disk_space_used(&self) -> u64 {
+        self.items.disk_space().get()
+    }
+
+    /// Performs maintenance, maybe deleting some old journals
+    #[inline(always)]
+    pub fn maintenance(&self) -> crate::Result<()> {
+        self.items.maintenance()
+    }
+
+    #[inline(always)]
+    pub fn rotate_journal(
+        &self,
+        journal_writer: &mut MutexGuard<Writer>,
+        watermarks: Vec<EvictionWatermark>,
+    ) -> crate::Result<()> {
+        self.items.rotate_journal(journal_writer, watermarks)
+    }
+}
+
+/// Write Buffer
+impl FlushTracker {
+    #[inline(always)]
+    pub fn buffer_size(&self) -> u64 {
+        self.tasks.buffer_size().get()
+    }
+
+    #[inline(always)]
+    pub fn grow_buffer(&self, n: u64) -> u64 {
```

Comment:
I'm trying to think of a better name for this because `grow_buffer` makes it sound like there is an actual buffer that is (re)allocated - the original `allocate` wasn't much better to be fair.

Reply:
For SpaceTracker it might be good to rename the methods to increment and decrement. Then grow/shrink_buffer becomes increment/decrement_buffer_size.

Reply:
Yeah that's better - also I think the #[inline] attribute can typically be omitted.